lib/toolbox/Taskfile.yaml

```diff
@@ -1,59 +1,96 @@
+  REGISTRY_LATEST_URL: "{{.REGISTRY_URL_BASE}}/{{.LATEST_IMAGE_TAG}}"
+  REGISTRY_VERSION_URL: "{{.REGISTRY_URL_BASE}}/{{.VERSION_IMAGE_TAG}}"
+  CONTAINERFILE_PATH: '{{.CONTAINERFILE_PATH | default "./Containerfile"}}'
   AWS_REGION: '{{.AWS_REGION | default "us-east-1"}}'
 
 tasks:
-  all:
+  auth:
+    desc: Authenticate to AWS ECR
+    internal: true
+    requires:
+      vars:
+        - AWS_REGION
+        - REGISTRY_URL_BASE
+    cmds:
+      - aws ecr get-login-password --region {{.AWS_REGION}} | docker login --username AWS --password-stdin {{.REGISTRY_URL_BASE}}
+
+  install:
+    desc: Install wrapper script from geodesic container
+    internal: true
+    requires:
+      vars:
+        - TOOLBOX_NAME
+    cmds:
+      - docker run --rm {{.REGISTRY_LATEST_URL}} init | bash -s {{.TOOLBOX_NAME}}
+
+  run:build:
     desc: Build, install, and run the toolbox image
     cmds:
       - task: build
-      - task: install
       - task: run
 
+  run:pull:
+    desc: Pull, install, and run the toolbox image
+    cmds:
+      - task: pull
+      - task: run
+
+  clean:
+    desc: Cleans all toolbox images off of the local machine
+    prompt: Are you sure you want to clean all {{.TOOLBOX_NAME}} toolbox images off of the local machine?
+    cmds:
+      - docker rmi $(docker images --format '{{ "{{" }}.Repository{{ "}}" }}:{{ "{{" }}.Tag{{ "}}" }}' | grep '{{.TOOLBOX_NAME}}')
+
   build:
     desc: Build and tag our toolbox image
     requires:
       vars:
-        - IMAGE_NAME
+        - TOOLBOX_NAME
         - CONTAINERFILE_PATH
     cmds:
-      - "docker build --tag {{.LATEST_IMAGE_TAG}} --tag {{.VERSION_IMAGE_TAG}} -f {{.CONTAINERFILE_PATH}} ."
+      - |
+        docker build --tag {{.LATEST_IMAGE_TAG}} \
+                     --tag {{.VERSION_IMAGE_TAG}} \
+                     --tag {{.REGISTRY_LATEST_URL}} \
+                     --tag {{.REGISTRY_VERSION_URL}} \
+                     -f {{.CONTAINERFILE_PATH}} .
       - echo "Done building and tagging toolbox 💯 Tagged as {{.VERSION_IMAGE_TAG}} + {{.LATEST_IMAGE_TAG}}"
 
-  install:
-    desc: Install wrapper script from geodesic container
-    requires:
-      vars:
-        - IMAGE_NAME
+  run:
+    desc: Run our toolbox image while also mounting your $HOME folder to `/localhost` in the container
+    deps:
+      - install
     cmds:
-      - docker run --rm {{.LATEST_IMAGE_TAG}} init | bash -s {{.IMAGE_NAME}} || (echo 'Try "sudo make install"'; exit 1)
+      - "{{.TOOLBOX_NAME}}"
 
   pull:
     desc: Pull our toolbox image from registry
+    deps:
+      - auth
     requires:
       vars:
         - REGISTRY_URL_BASE
-        - REGISTRY_URL_FULL
+        - REGISTRY_LATEST_URL
     cmds:
-      - |
-        aws ecr get-login-password --region $(AWS_REGION) | docker login --username AWS --password-stdin {{.REGISTRY_URL_BASE}}
-        docker pull {{.REGISTRY_URL_FULL}}
+      - docker pull {{.REGISTRY_LATEST_URL}}
+      - docker tag {{.REGISTRY_LATEST_URL}} {{.LATEST_IMAGE_TAG}}
```

Comment:
move these higher than `run:pull` and `run:build` so they can be read chronologically

Reply:
Good call -- will do 👍